publicsuffixlist.js

```diff
@@ -242,17 +242,19 @@ class PublicSuffixList {
 
                 // Ignore surrounding whitespaces
                 line = line.trim();
-                if ( line.length === 0 ) { continue; }
 
-                const exception = line.charCodeAt(0) === 0x21 /* '!' */;
+                const exception = line.length > 0 && line.charCodeAt(0) === 0x21 /* '!' */;
                 if ( exception ) {
                     line = line.slice(1);
                 }
 
-                if ( mustPunycode.test(line) ) {
+                if ( line.length > 0 && mustPunycode.test(line) ) {
                     line = toAscii(line.toLowerCase());
                 }
 
+                // https://en.wikipedia.org/wiki/Hostname#Syntax
+                if ( line.length === 0 || line.length > 253 ) { continue; }
```

Comment:
The `=== 0` check is needed one more time because `url.domainToASCII()` can return a blank string.

Reply:
Minor: I use `!== 0` instead of `> 0` for testing string lengths against 0, I would prefer to keep code consistent here.

Reply:
Why not just move the `if ( line.length === 0 ) { continue; }` after the test for exception? This will simplify both following tests.